misc/pytorch_toolkit/breast_ultrasound_simulation/src/model.py

```diff
@@ -0,0 +1,419 @@
+
+        self.down2 = nn.Sequential(
+            # 1x1 for channel change
+            nn.Conv2d(32, 16, 1, padding=0, stride=1, bias=True),
+            nn.BatchNorm2d(16),
+            nn.LeakyReLU(inplace=True),
+            # 3x3 spectral conv
+            SpectralConv2dInter(
+                16, k=3, pad=2, dil=2, wts_list=model.down2[3], a=a),
+            nn.BatchNorm2d(16),
+            nn.LeakyReLU(inplace=True),
+            # 1x1 for channel change
+            nn.Conv2d(16, 1, 1, padding=0, stride=1, bias=True),
+            nn.BatchNorm2d(1),
+            nn.LeakyReLU(inplace=True),
+            # 3x3 spectral conv
+            SpectralConv2d(1, k=1, p=0, py=0, s=1, dil=1),
+            nn.Sigmoid()
+        )
+
+    def forward(self, x):
+        x = self.down1(x)
+        x = self.res1(x)
+        x = self.res2(x)
+        x = self.res3(x)
+        x = self.res4(x)
+        x = self.down2(x)
+
+        return x
+
+
+class Generator3dInter(nn.Module):
+    def __init__(self, in_ch):
+        super().__init__()
+
+        self.down1 = nn.Sequential(
+            # 1x1 for channel change
+            nn.Conv3d(in_ch, 16, 1, padding=0, stride=1, bias=True),
+            nn.BatchNorm3d(16),
+            nn.LeakyReLU(inplace=True),
+            # 3x3 spectral conv
+            SpectralConv3dInter(
+                16, k=3, pad=2, dil=2),
+            nn.BatchNorm3d(16),
+            nn.LeakyReLU(inplace=True),
+            # 1x1 for channel change
+            nn.Conv3d(16, 32, 1, padding=0, stride=1, bias=True),
+            nn.BatchNorm3d(32),
+            nn.LeakyReLU(inplace=True),
+            # 3x3 spectral conv
+            SpectralConv3dInter(32, k=3, pad=2, dil=2),
+            nn.BatchNorm3d(32),
+            nn.LeakyReLU(inplace=True)
+        )
+
+        self.res1 = ResidualBlock3dInter(32, py=2, dil=2)
+        self.res2 = ResidualBlock3dInter(32, py=2, dil=2)
+        self.res3 = ResidualBlock3dInter(32, py=2, dil=2)
+        self.res4 = ResidualBlock3dInter(32, py=2, dil=2)
+
+        self.down2 = nn.Sequential(
+            # 1x1 for channel change
+            nn.Conv3d(32, 16, 1, padding=0, stride=1, bias=True),
+            nn.BatchNorm3d(16),
+            nn.LeakyReLU(inplace=True),
+            # 3x3 spectral conv
+            SpectralConv3dInter(
+                16, k=3, pad=2, dil=2),
+            nn.BatchNorm3d(16),
+            nn.LeakyReLU(inplace=True),
+            # 1x1 for channel change
+            nn.Conv3d(16, 1, 1, padding=0, stride=1, bias=True),
+            nn.BatchNorm3d(1),
+            nn.LeakyReLU(inplace=True),
+            # 3x3 spectral conv
+            SpectralConv3dInter(1, k=1, pad=0, dil=1),
+            nn.Sigmoid()
+        )
+
+    def forward(self, x):
+        x = self.down1(x)
+        x = self.res1(x)
+        x = self.res2(x)
+        x = self.res3(x)
+        x = self.res4(x)
+        x = self.down2(x)
+
+        return x
```

Comment:
Try to move common parts of the Generator model classes to one super-class. For example, `forward` method is the same and, I guess, some parts of the `__init__` can be unified also.